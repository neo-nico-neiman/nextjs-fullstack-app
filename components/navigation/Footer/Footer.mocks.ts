import { IFooter } from './Footer';

const base: IFooter = { 
    // add props
};

const alt: IFooter = {
    // add props
};

export const mockFooterProps = {
    base,
    alt,
};