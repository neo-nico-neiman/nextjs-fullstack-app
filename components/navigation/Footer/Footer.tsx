import Link from 'next/link';

export interface IFooter extends React.ComponentPropsWithoutRef<'footer'> {}

const Footer: React.FC<IFooter> = ({ className, ...footerProps }) => {
  return (
    <footer
      {...footerProps}
      className={`w-full px-5 py-2 bg-slate-100 text-slate-500 flex justify-between items-center flex-col sm:flex-row ${className}`}
    >
      <div className="space-x-5 m-5">
        <Link href="/">
          <a className="hover:underline">Advertising</a>
        </Link>
        <Link href="/">
          <a className="hover:underline">Business</a>
        </Link>
        <Link href="/">
          <a className="hover:underline">How Search Works</a>
        </Link>
      </div>

      <div className="space-x-5 m-5">
        <Link href="/">
          <a className="hover:underline sm:inline">Privacy</a>
        </Link>
        <Link href="/">
          <a className="hover:underline sm:inline">Terms</a>
        </Link>
        <Link href="/">
          <a className="hover:underline sm:inline">Settings</a>
        </Link>
      </div>
    </footer>
  );
};

export default Footer;
