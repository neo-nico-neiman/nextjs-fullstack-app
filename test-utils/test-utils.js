// test-utils.js
import Providers from '@state/providers';
import { render } from '@testing-library/react';

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
