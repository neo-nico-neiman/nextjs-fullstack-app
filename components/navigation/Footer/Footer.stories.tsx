import { ComponentMeta, ComponentStory } from '@storybook/react';

import Footer from './Footer';

export default {
  title: 'Navigation/Footer',
  component: Footer,
} as ComponentMeta<typeof Footer>;

const Template: ComponentStory<typeof Footer> = (args) => <Footer {...args} />;
Template.args = {};

export const Example = Template.bind({});
Example.args = {};
