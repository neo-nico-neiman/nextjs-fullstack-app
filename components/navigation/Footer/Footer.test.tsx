import { render, screen } from '@test-utils/test-utils';
import '@testing-library/jest-dom';

import Footer from './Footer';

describe('Footer', () => {
  test('Should match the snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment).toMatchSnapshot();
  });

  test('Should contain links', () => {
    render(<Footer />);
    expect(
      screen.getByRole('link', { name: /advertising/i })
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /business/i })).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /how search works/i })
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /privacy/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /terms/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /settings/i })).toBeInTheDocument();
  });
});
