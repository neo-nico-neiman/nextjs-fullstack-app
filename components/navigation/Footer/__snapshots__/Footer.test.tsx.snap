// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should match the snapshot 1`] = `[Function]`;
